cpp/include/cudf/functions.h

```diff
@@ -796,6 +796,36 @@ gdf_error gdf_min(gdf_column *col, void *dev_result, gdf_size_type dev_result_si
  * --------------------------------------------------------------------------*/
 gdf_error gdf_max(gdf_column *col, void *dev_result, gdf_size_type dev_result_size);
 
+/* --------------------------------------------------------------------------*
+ * @brief  Computes the rolling window function of the values in a column. This
+ * is matching Pandas' API for DataFrame.rolling with one notable difference -
+ * instead of the center flag, in cuDF we're using the forward window size
+ * which allows for more flexible windows. Note, that the total window size is
+ * window + forward_window. The element i uses [i-window+1, i+forward_window]
+ * elements to do the window computation.
+ *
+ * @param[out] output_col The output column
+ * @param[in] input_col The input column
+ * @param[in] window The rolling window size, output_col[i] will accumulate
+ *                values from input_col[i-window+1] to input_col[i] inclusive
+ * @param[in] min_periods Minimum number of observations in window required to
```

Comment:
Do we really want to store NaN? Is that what Pandas does? NaNs are viral... might not be desirable.

Reply:
I agree with you. I updated the description to clarify that we'll use the valid bit mask instead. @kkraus14 is onboard with this